### ISSUE TITLE
Query: Data Formatting for detection history.

### ISSUE THREAD

**Issue by rahulkgour** [doserjef/spOccupancy#37]
Hi Jeff,

First I would like to thank you for the great package considering its application. 

I am trying to manage a multi season single species spatio-temporal analysis.

I am reaching out to you with a query rather than an issue. I have begun compiling input data as per the requirements of the package. However, I am encountering an issue in creating an array in the required format for the detection history. I have successfully compiled the rest of the data for detection covariates, site covariates, and coordinates as explained in the vignette.

For your reference, I have attached a sample dataset for the detection history. 

First approach while using the attached sample dataset: ([NewDetHistory.csv](https://github.com/doserjef/spOccupancy/files/14323077/NewDetHistory.csv)): I created a data frame with plots as rows and replicates as columns, and then attempted to convert this into a 2-dimensional array. Unfortunately, this did not work.

Second approach - creating a data frame after removing the Plot Ids from the dataset and then converting into an array, mentioned below as Array1.

<img width="318" alt="Screenshot 2024-02-18 at 21 46 15" src="https://github.com/doserjef/spOccupancy/assets/39184822/99368e81-d8b4-4528-80b5-b94b90188254">

In this data set, we have 70 plots and for each plot we have 10 replicates.

I would greatly appreciate any guidance or assistance you could provide on this matter.

Thank you,
Rahul

**Comment by rahulkgour** [comment]
Hi,

I have created the dataset with only few variables to give a test run, hope its accordance to the specified requirements.

All the data into list object.

<img width="548" alt="Screenshot 2024-02-19 at 00 28 55" src="https://github.com/doserjef/spOccupancy/assets/39184822/284542ee-c450-41dc-a4a9-e5e4e4ed47b4">

Now when I use the above data set and calling the function "spMsPGOcc" I am getting the error as, "**Error in y[, ord, , drop = FALSE] : subscript out of bounds**".

Thanks.

**Comment by rahulkgour** [comment]
Hi Jeff, 

I am really sorry about posting the things in the same ticket, as I think its mostly due to formatting of data especially for detection history.

Now, I modified the y array further to rectify the "script out of bounds" error, but now I have got another error instead. 

<img width="663" alt="Screenshot 2024-02-19 at 10 33 07" src="https://github.com/doserjef/spOccupancy/assets/39184822/c2e5b370-b36e-4548-8e2f-810571f63fb9">


Seems like this error is due to the dimensions of out$z.samples which is expecting to be a 2-dimensional array, but it seems to be a 3-dimensional array in my case.

I pasted the dimensions of y, occ.covs, det.covs, and coords for your reference, let me know if any other information needed. I believe I am stuck now.

<img width="687" alt="Screenshot 2024-02-19 at 10 43 48" src="https://github.com/doserjef/spOccupancy/assets/39184822/151fc834-c8f1-4158-8600-e14ab716db89">


Thanks,
Rahul

**Comment by rahulkgour** [comment]
Hi Jeff, I should really apologies for too many edits and messages. 

I found the issues now. 
1. For single species analysis, I was unnecessary creating a 3D array, which I corrected;    
2. I was calling a wrong function, I was triggering multi-species occupancy function, I had to call 'spPGOcc' for single species, which I corrected and it worked fine, now I can use my actual data and run the analysis. 

Sorry again if any inconvenience caused, I am closing this ticket.
